### ISSUE TITLE
(en) Requests Implementations - (es) Implementaciones de solicitudes

### ISSUE THREAD

**Issue by Alexandro2000** [eusonlito/Password-Manager#9]
EN:
Requests Implementations: - group and user import with LDAP protocol
- Authentication with protocol
 LDAP

ES:
Implementaciones de solicitudes: - importación de grupos y usuarios con protocolo LDAP
- Autenticación con protocolo
 LDAP

**Comment by eusonlito** [comment]
Lidiar con la gestión y sincronización de usuarios vía LDAP es siempre un lío importante.

Una posibilidad sería crear una serie de endpoints API que permitan realizar integraciones personalizadas en Equipos y Usuarios.

**Comment by Alexandro2000** [comment]
Tu idea es muy buena.
Sin embargo, tan pronto como tengo un momento, lo implemento de todos modos. Recordemos que al cliente le interesa un producto terminado con la mayor cantidad de funcionalidades posible.

También tu proyecto está muy bien hecho, felicitaciones.
quiero ayudar a que crezca

**Comment by eusonlito** [comment]
Sí, en cualquier caso existen muchas implicaciones diferentes en LDAP y la gestión es compleja y muy de detalles.

Yo creo que sería mucho más útil crear una especie conector externo que sincronice un sistema LDAP a la plataforma usando los nuevos endpoints API que se crearían.

Así por ejemplo se podrían crear tantos conectores como fueran necesarios para diferentes sistemas de autenticación externos (Google, LDAP, etc...).

Es mejor un core más enfocado a la funcionalidad principal y extenderlo desde fuera con otras features, así también permitiría un mantenimiento (y actualización) mucho más sencillo en varios repositorios.

**Comment by eusonlito** [comment]
Se ha añadido una serie de endpoints que permiten la administración de usuarios con las siguientes rutas:

https://github.com/eusonlito/Password-Manager/blob/85408d5f3e6186c41bf3d299c1e2b8c895bbe382/app/Domains/User/ControllerApi/router.php#L7-L14

Para poder acceder a estas es necesario ser usuario administrador.

La autenticación se realiza con el API KEY de usuario en la cabecera `Authorization` y el valor de `api_secret` en el campo `POST`,

Los endpoints son siempre `POST`:

* Listado de usuarios: `/user/api/index`
* Detalle de usuario: `/user/api/{id}/detail`
* Alta de usuario: `/user/api/create` con validación https://github.com/eusonlito/Password-Manager/blob/master/app/Domains/User/Validate/Create.php
* Actualización de usuario: `/user/api/{id}/update` con validación https://github.com/eusonlito/Password-Manager/blob/master/app/Domains/User/Validate/Update.php
* Actualización de Equipos de usuario: `/user/api/{id}/update/team` con validación https://github.com/eusonlito/Password-Manager/blob/master/app/Domains/User/Validate/UpdateTeam.php
* Cambio de estado de usuario (activo/desactivo): `/user/api/{id}/update/boolean/{column}` con validación https://github.com/eusonlito/Password-Manager/blob/master/app/Domains/User/Validate/UpdateBoolean.php
